ethylene-core/src/main/java/com/github/steanky/ethylene/core/processor/ConfigProcessor.java

```diff
@@ -0,0 +1,27 @@
+package com.github.steanky.ethylene.core.processor;
+
+import com.github.steanky.ethylene.core.ConfigElement;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Processes some configuration data. Fundamentally, implementations of this interface act as simple bidirectional
+ * mapping functions between {@link ConfigElement} instances and arbitrary data.
+ * @param <TData> the type of data to convert to and from
+ */
+public interface ConfigProcessor<TData> {
+    /**
+     * Produces some data from a provided {@link ConfigElement}.
+     * @param element the element to process
+     * @return the data object
+     * @throws ConfigParseException if the provided {@link ConfigElement} does not contain valid data
+     */
+    @NotNull TData dataFromElement(@NotNull ConfigElement element) throws ConfigParseException;
+
+    /**
+     * Produces a {@link ConfigElement} from the provided data object.
+     * @param data the data object
+     * @return a {@link ConfigElement} representing the given data
+     * @throws ConfigParseException if the data is invalid
```

Comment:
How can data be invalid?

Reply:
implementation defined (maybe users want to prevent values in a certain range from being written to the file, or the data contains an object that can't be converted to a ConfigElement)

Reply:
i mostly just added it for symmetry with dataFromElement though